src/UnderlineNav2/UnderlineNav.tsx

```diff
@@ -235,6 +243,46 @@ export const UnderlineNav = forwardRef(
       // eslint-disable-next-line no-console
       console.warn('Use the `aria-label` prop to provide an accessible label for assistive technology')
     }
+    const [isWidgetOpen, setIsWidgetOpen] = useState(false)
+
+    const closeOverlay = React.useCallback(() => {
+      setIsWidgetOpen(false)
+    }, [setIsWidgetOpen])
+
+    const toggleOverlay = React.useCallback(() => {
+      setIsWidgetOpen(!isWidgetOpen)
+    }, [setIsWidgetOpen, isWidgetOpen])
+
+    const focusOnMoreMenuBtn = React.useCallback(() => {
+      moreMenuBtnRef.current?.focus()
+    }, [])
+
+    useFocusZone({
+      containerRef: backupRef,
+      bindKeys: FocusKeys.ArrowVertical | FocusKeys.ArrowHorizontal | FocusKeys.HomeAndEnd | FocusKeys.Tab
+    })
+
+    useOnEscapePress(
+      (event: KeyboardEvent) => {
+        if (isWidgetOpen) {
+          event.preventDefault()
+          closeOverlay()
+          focusOnMoreMenuBtn()
+        }
+      },
+      [isWidgetOpen]
+    )
+
+    useOnOutsideClick({onClickOutside: closeOverlay, containerRef, ignoreClickRefs: [moreMenuBtnRef]})
+    const onAnchorClick = useCallback(
+      (event: React.MouseEvent<HTMLButtonElement>) => {
+        if (event.defaultPrevented || event.button !== 0) {
+          return
+        }
+        toggleOverlay()
```

Comment:
This seems to be the only place where `toggleOverlay` is called, would it make sense to inline it from L:252 here?

```suggestion
        setIsWidgetOpen(isWidgetOpen => !isWidgetOpen)
```

Reply:
Yes, it does! 🙂 Thank you 🙌🏼 I'll do add the commit on my local as I also need to remove the toggleOverlay function ✨